src/lightgbm/src/main/scala/LightGBMRegressor.scala

```diff
@@ -120,4 +123,22 @@ class LightGBMRegressionModel(override val uid: String, model: LightGBMBooster,
   def getModel: LightGBMBooster = this.model
 }
 
-object LightGBMRegressionModel extends ConstructorReadable[LightGBMRegressionModel]
+object LightGBMRegressionModel extends ConstructorReadable[LightGBMRegressionModel] {
+  def loadNativeModelFromFile(session: SparkSession, filename: String, labelColName: String = "label",
+                              featuresColName: String = "features",
+                              predictionColName: String = "prediction"): LightGBMRegressionModel = {
+    val uid = Identifiable.randomUID("LightGBMRegressor")
+    val textRdd = session.read.text(filename)
```

Comment:
Could you use SparkSession.getOrCreate() for a cleaner api here?

Reply:
done!